sdk/eventhub/Azure.Messaging.EventHubs/stress/src/Infrastructure/BufferedPublisher.cs

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Messaging.EventHubs.Consumer;
+using Azure.Messaging.EventHubs.Producer;
+using Azure.Messaging.EventHubs.Tests;
+
+namespace Azure.Messaging.EventHubs.Stress
+{
+    internal class BufferedPublisher
+    {
+        private string connectionString;
+        private string eventHubName;
+        private Metrics metrics;
+        private ProducerConfiguration testConfiguration;
+
+        public BufferedPublisher(ProducerConfiguration testConfigurationIn, Metrics metricsIn)
+        {
+            connectionString = testConfigurationIn.EventHubsConnectionString;
+            eventHubName = testConfigurationIn.EventHub;
+            testConfiguration = testConfigurationIn;
+            metrics = metricsIn;
+        }
+
+        public async Task Start(CancellationToken cancellationToken)
+        {
+            var enqueueTasks = new List<Task>();
+
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                using var backgroundCancellationSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+
+                try
+                {
+                    var producer = new EventHubBufferedProducerClient(connectionString, eventHubName);
```

Comment:
We'll definitely want to refactor a bit to make this class (and similar role-type things) async disposable;  we need to be able to dispose the producer without flushing as part of cleanup.